### ISSUE TITLE
C-API : How to run classification

### ISSUE THREAD

**Issue by astariul** [dmlc/xgboost#10338]
I'm using the C-API from C++. First of all, thank you for the great documentation with all the code examples, this is insanely helpful.

My problem is a classification problem. But in [C-API tutorial](https://xgboost.readthedocs.io/en/stable/tutorials/c_api_tutorial.html#useful-tips-to-remember), the example given is for a regression : 

```
// variable to store labels for the dataset created from above matrix
float labels[ROWS];

for (int i = 0; i < ROWS; i++) {
  labels[i] = i;
}

// Loading the labels
safe_xgboost(XGDMatrixSetFloatInfo(dmatrix, "label", labels, ROWS));
```

And later, when predicting : 

```
for (unsigned int i = 0; i < output_length; i++){
  printf("prediction[%i] = %f \n", i, output_result[i]);
}
```

The output is a float.

---

I'm trying to change the code to deal with a classification problem. I could successfully create the labels as `uint` : 

```
  // variable to store labels for the dataset created from above matrix
  uint32_t labels[ROWS];

  for (uint32_t i = 0; i < ROWS; i++) {
    labels[i] = i;
  }
  // Loading the labels
  XGDMatrixSetUIntInfo(dtrain, "label", labels, ROWS);
```

But when running the predictions, I still get a float prediction. I'm not sure how to change the prediction code to get "classes" instead (`uint`). Any pointer to share ?

**Comment by trivialfis** [comment]
The document for C is mostly geared toward others who want to work on higher-level language interfaces, so the machine-learning aspect of the document is relatively lacking. The best place to start at the moment is the Python interface and work your way back to C, where the former is completely defined on top of the latter without any helper C code.

That said, you need to:
- Provide the objective and the number of classes `num_class` to XGBoost during training.
- Define labels as classes:
```
[1, 0, 3, 2, 1, 2]
```
This is a set of labels with 6 samples, with a total of 4 classes in the dataset.
- For class output instead of probability output, use the `multi:softmax` objective instead of the `multi:softprob` objective. Otherwise, you can calculate the estimated class labels yourself based on the estimated probability.

**Comment by trivialfis** [comment]
Feel free to reopen if there are further questions.
